paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -209,9 +209,34 @@ internal class CustomerSheetViewModel @Inject constructor(
         // TODO (jameswoo) handle onFormValuesChanged
     }
 
-    @Suppress("UNUSED_PARAMETER")
     private fun onItemRemoved(paymentMethod: PaymentMethod) {
-        TODO()
+        viewModelScope.launch {
+            paymentMethod.id?.let { paymentMethodId ->
+                customerAdapter.detachPaymentMethod(paymentMethodId = paymentMethodId)
+                    .onFailure { throwable ->
+                        updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
+                            it.copy(
+                                errorMessage = throwable.message,
+                            )
+                        }
+                    }.onSuccess { paymentMethod ->
+                        updateViewState<CustomerSheetViewState.SelectPaymentMethod> { previousViewState ->
+                            val paymentMethodSelection = previousViewState.paymentSelection as? PaymentSelection.Saved
+                            val previousPaymentMethods = previousViewState.savedPaymentMethods.toMutableList()
+                            val paymentMethodToRemove = previousPaymentMethods.find { it.id == paymentMethodId }
+                            previousPaymentMethods.remove(paymentMethodToRemove)
+                            previousViewState.copy(
+                                savedPaymentMethods = previousPaymentMethods,
+                                paymentSelection = if (paymentMethodSelection?.paymentMethod == paymentMethod) {
+                                    null
+                                } else {
+                                    previousViewState.paymentSelection
+                                }
+                            )
+                        }
```

Comment:
Nit: Can make this a bit more succinct.

```suggestion
                            viewState.copy(
                                savedPaymentMethods = viewState.savedPaymentMethods.filter { pm ->
                                    pm.id != paymentMethodId
                                },
                                paymentSelection = viewState.paymentSelection.takeUnless { selection ->
                                    selection is PaymentSelection.Saved &&
                                        selection.paymentMethod == paymentMethod
                                },
                            )
```